lib/quicly.c

```diff
@@ -3396,8 +3365,6 @@ static int handle_ack_frame(quicly_conn_t *conn, size_t epoch, quicly_ack_frame_
         packet_number += frame->gaps[gap_index];
     }
 
-    LOG_CONNECTION_EVENT(conn, QUICLY_EVENT_TYPE_QUICTRACE_RECV_ACK, INT_EVENT_ATTR(ACK_DELAY, frame->ack_delay));
-
```

Comment:
QUICTRACE_RECV_ACK seems to have two signatures (i.e. output parameters; see above and here).

Not knowning which one is correct, I have tentatively remove this invocation.

Reply:
They're both correct. The previous signature is used for logging ack blocks, which can be multiple per ack frame, and this one is used for logging the ack delay, which is once per ack frame.  The post processing script was using the attribute (ACK_DELAY, ACK_BLOCK_BEGIN) to demux trace outputs of type RECV_ACK. 

That was also being used as a marker for the end of ack processing. I think the simplest mapping is to add a new probe for RECV_ACK_DELAY here.

Reply:
Related question: what's the unit of ack_delay to be logged? I assume we are expected to normalize the value instead of just emitting what's on the wire.

Reply:
It's currently being logged in microsecs.

Reply:
IIUC what's currently used is the value contained in the ACK frame. Therefore the unit would depend on the peer. We normalize the value in the `quicly_loss_on_ack_received` that is being called below.